compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -30,33 +30,55 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Bit.hpp"
 
-static TR::Register *addOrSubInteger(TR::Node *node, TR::CodeGenerator *cg)
+static TR::Register *
+addOrSub32Or64Integer(TR::Node *node, TR::CodeGenerator *cg)
    {
    TR::Node *firstChild = node->getFirstChild();
-   TR::Register *src1Reg = cg->evaluate(firstChild);
+   /* Get a clobberable register (re-uses one if possible otherwise allocates one) */
+   TR::Register *src1Reg = cg->gprClobberEvaluate(firstChild);
```

Comment:
A clobber evaluation for both of these source registers is not what you want here.  A clobber evaluation is useful when the register itself will be modified by the instruction but yet it's original value is still needed (because of commoning, for example).  Clobber evaluation will insert an instruction to copy the register to a temporary register if its reference count is > 1 (i.e., its value is still consumed elsewhere) and skip the copy if this is the last reference it.

Simple calls to `evaluate` are what you want here.

However, there is an opportunity to optimize the target register though.  If one of the source registers "dies" at this instruction then you can re-use it for the target register rather than allocating a new register.  So, if the reference count on either source register == 1 then you can conclude that this is the last reference to that node and its register live range ends here and can be reused.

Reply:
Ok fair enough. Agreed. I'll check the reference count at this point to see about re-use.